sw/device/sca/lib/aes.c

```diff
@@ -0,0 +1,237 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
```

Comment:
It's important to specify here what this AES implementation is about. No one should get the impression that this is production code we intend to run on Ibex to perform AES encryption/decryption or random number generation in the field. First of all, we do have hardened hardware primitives for both tasks. Second, we will never harden this code in a similar way we harden other production code in the repo. So no-one in the project or external should get tempted to use this for their applications / libraries.

Instead, it should be clearly stated here that the only purpose of this implementation is to serve as PRNG for SCA experiments and penetration testing where can't use the AES/CSRNG hardware primitives because we're analyzing the hardware.

Reply:
Good idea,
I've added a note to clarify this.